packages/solarwinds-apm/src/web/init.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader,
+} from "@opentelemetry/sdk-metrics"
+import {
+  BatchSpanProcessor,
+  WebTracerProvider,
+} from "@opentelemetry/sdk-trace-web"
+import { ATTR_SERVICE_NAME } from "@opentelemetry/semantic-conventions"
+import {
+  getInstrumentations,
+  getResource,
+} from "@solarwinds-apm/instrumentations/web"
+import { BucketType, Flags, SampleSource } from "@solarwinds-apm/sampling"
+
+import { LogExporter } from "../exporters/logs.js"
+import { MetricExporter } from "../exporters/metrics.js"
+import { TraceExporter } from "../exporters/traces.js"
+import { LocationProcessor } from "../processing/location.js"
+import { ParentSpanProcessor } from "../processing/parent-span.js"
+import { ResponseTimeProcessor } from "../processing/response-time.js"
+import { TransactionNameProcessor } from "../processing/transaction-name.js"
+import { RequestHeadersPropagator } from "../propagation/headers.js"
+import { TraceContextPropagator } from "../propagation/trace-context.js"
+import { HttpSampler } from "../sampling/http.js"
+import { SAMPLER } from "../shared/init.js"
+import { componentLogger } from "../shared/logger.js"
+import { VERSION } from "../version.js"
+import { type Configuration, read } from "./config.js"
+import { Logger } from "./logger.js"
+
+export function init() {
+  try {
+    const config = read()
+
+    diag.setLogger(new Logger(), config.logLevel)
+    const logger = componentLogger(init)
+    logger.debug("config", config)
+
+    const resource = Resource.default()
+      .merge(
+        new Resource({
+          [ATTR_SERVICE_NAME]: config.service,
+          "sw.data.module": "apm",
+          "sw.apm.version": VERSION,
+        }),
+      )
+      .merge(getResource())
+
+    const meterProvider = initMetrics(config, resource)
+    const tracerProvider = initTracing(config, resource)
+    initLogs(config, resource)
+
+    registerInstrumentations({
+      instrumentations: getInstrumentations(),
+      tracerProvider,
+      meterProvider,
+    })
+
+    logger.debug("resource", resource.attributes)
+  } catch (error) {
+    console.error("solarwinds-apm", error)
+  }
+}
+
+function initTracing(config: Configuration, resource: Resource) {
+  const sampler = new HttpSampler(config, {
+    sampleRate: 1_000_000,
+    sampleSource: SampleSource.LocalDefault,
+    flags: Flags.SAMPLE_START | Flags.SAMPLE_THROUGH_ALWAYS,
+    buckets: {
+      [BucketType.DEFAULT]: {
+        capacity: 10,
+        rate: 0,
+      },
+    },
+    timestamp: Math.round(Date.now() / 1000),
+    ttl: 10,
+  })
+  SAMPLER.resolve(sampler)
+
+  const provider = new WebTracerProvider({
+    resource,
+    sampler,
+    spanProcessors: [
+      new TransactionNameProcessor(config),
+      new ResponseTimeProcessor(),
+      new BatchSpanProcessor(new TraceExporter(config)),
+      new ParentSpanProcessor(),
+      new LocationProcessor(),
+    ],
+  })
+  provider.register({
```

Comment:
Another dumb question: Is there still an APM JS response propagator, and if so is it needed?

Reply:
There is, and it's not required since browsers only ever make requests

Reply:
🤯